src/frequenz/sdk/timeseries/_ringbuffer/buffer.py

```diff
@@ -257,8 +257,49 @@ def to_internal_index(
             )
         )
 
+    def index_to_datetime(self, index: int) -> datetime | None:
```

Comment:
Since we stripped the *suffix* in other methods (`to_internal_index()`), I wonder if we couldn't do the same here, so `to_datetime()`. This is basically getting the timestamp of the value stored at `index`, right? In that case it might even be `get_timestamp()`.

I also noticed that sometimes we are using `datetime` to refer to the timestamp (like `_datetime_newest`), I think we should use `timestamp` consistently and leave only `datetime` for the time, as we are really storing **timestamps** here, which happen to be represented by a `datetime` object (but it could be a `float` or something else too).

Reply:
Makes sense, made it `get_timestamp`. Now from outside there are no `datetime` methods, only private `_datetime_{oldest,newest}` in the ring buffer. I leave those for now, but if you think we should still change it I could create an issue for this?

Reply:
https://github.com/frequenz-floss/frequenz-sdk-python/issues/691